### PR TITLE
Redesign ham10k

### PR DESCRIPTION
**Pull request by CengizHenk**

### CODE REVIEW

File: src/outlier_hub/datasets/ham10k/factory.py [hunk 1]
```diff
@@ -1,75 +1,152 @@
...
-# Code for testing the dataset
 if __name__ == "__main__":
+    
     # get root workind directory path
-    root_path = pathlib.Path.cwd().parents[3]
+    root_path = pathlib.Path.cwd()
+    logger.debug(f' pathlib.Path.cwd(): { pathlib.Path.cwd()}')
 
     # complete path to manual added data
     data_path = os.path.join(root_path, "src/outlier_hub/datasets/ham10k/data")
```
Comment (le1nux) [inline src/outlier_hub/datasets/ham10k/factory.py:89]: Can you specify another path here, please? src is just meant for sourcecode

File: src/outlier_hub/datasets/ham10k/factory.py [hunk 2]
```diff
@@ -1,75 +1,152 @@
...
+            c = c + 1
+        elif int(target) == 3:
+            d = d + 1
+        elif int(target) == 4:
+            e = e + 1
+        elif int(target) == 5:
+            f = f + 1
+        elif int(target) == 6:
+            g = g + 1
+    print('verteilung:',a,b,c,d,e,f,g)
```
Comment (le1nux) [inline src/outlier_hub/datasets/ham10k/factory.py:128]: for this we have collections count in python

File: src/outlier_hub/datasets/ham10k/preprocessor.py [hunk 3]
```diff
@@ -1,120 +1,106 @@
...
+            streamed_resource = self.storage_connector.get_resource(prep_dataset_identifier)
+
         return streamed_resource
+
+    def prepare_dataset(self,
+                        h5py_file : h5py.File,
+                        raw_samples_identifier,
+                        raw_targets_identifier):
+
+        df = pd.read_csv('src/outlier_hub/datasets/ham10k/data/raw/train/train_labels/ISIC2018_Task3_Training_GroundTruth.csv', header=0)
```
Comment (le1nux) [inline src/outlier_hub/datasets/ham10k/preprocessor.py:81]: I would prefer a path outside of src

**Review by le1nux — CHANGES_REQUESTED** [review]
Overall it looks good to me :+1: Just left some minor comments. 
Maybe I missed something, but did you add the preprocessor and iteraor for the ODIR dataset?

File: src/outlier_hub/datasets/ham10k/factory.py (hunk 1 above)
Comment (ichbk) [reply src/outlier_hub/datasets/ham10k/factory.py]: Of course - Done

File: src/outlier_hub/datasets/ham10k/factory.py (hunk 2 above)
Comment (ichbk) [reply src/outlier_hub/datasets/ham10k/factory.py]: Thanks for the hint - Done

File: src/outlier_hub/datasets/ham10k/preprocessor.py (hunk 3 above)
Comment (ichbk) [reply src/outlier_hub/datasets/ham10k/preprocessor.py]: Of course - Done

**Comment by ichbk** [comment]
Here are some comflicts, which I could resolve. When I do that, it only offers to merge at the end. So I would like to get your OK, for that.

ODR implementation is included in this branch, not sure why it is not clearly visible.

**Pull request merged**